### PR TITLE
feat(telecom): delete all softphone devices

### PR DESCRIPTION
**Pull request by antonymarion**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat#MANAGER-14466
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
~~- [ ] Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.html [hunk 1]
```diff
@@ -3,10 +3,17 @@
     data-on-dismiss="$ctrl.goBack()"
     data-primary-action="$ctrl.deleteDevice()"
     data-primary-label="{{:: 'telephony_line_softphone_btn_delete' | translate }}"
-    data-heading="{{ ::'telephony_line_softphone_delete_device_modal_title' | translate }}"
+    data-heading="{{ ($ctrl.deleteAllDevices ? 'telephony_line_softphone_delete_all_devices_modal_title' : 'telephony_line_softphone_delete_device_modal_title') | translate }}"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.html:6]:
```suggestion
    data-heading="{{ $ctrl.deleteAllDevices ? 'telephony_line_softphone_delete_all_devices_modal_title' : 'telephony_line_softphone_delete_device_modal_title' | translate }}"
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 2]
```diff
@@ -192,6 +192,17 @@ <h4 class="pt-3">
     >
         <span data-translate="telephony_line_softphone_add_device"></span>
     </oui-button>
+    <oui-button
+        class="ml-3 mt-3"
+        data-ng-if="$ctrl.softphoneDevices.length===3"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:197]:
```suggestion
        data-ng-if="$ctrl.softphoneDevices.length === 3"
```

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]
Instead of reusing the existing `delete`, create a new folder `delete-all` with its own module, routing. Maybe you can move the component definition to a common place and reuse it in both actions.
Each new route will require a new folder.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/index.js [hunk 3]
```diff
@@ -5,24 +5,21 @@ import routing from './softphone.routing';
...
-    sendLinkModule,
-    deleteDeviceModule,
     ngOvhTelecomUniverseComponents,
     'ngTranslateAsyncLoader',
     'pascalprecht.translate',
     'ui.router',
+    'oc.lazyLoad',
   ])
   .component('themePicker', themeComponent)
+  .component('deleteModal', deleteModalComponent)
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/index.js:22]:
```suggestion
  .component('telecomTelephonyLineSoftphoneDeleteModal', deleteModalComponent)
```

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**